docs/assignment2.md

```diff
@@ -1 +1,233 @@
 # Assignment 2
+
+### Implemented feature
+
+| ID  | Short name  | Description  |
+|---|---|---|
+| F6  | League Formation | Allow the user to set up a league with its own schedule. The first round can have at most 16 matches. Matches will be played according to this schedule and the system will output the winner of each individual game. |
+
+This is the only feature we implemented; a league has (for now) a fixed-size of 16 teams in the first round. Players and teams are hard-coded into the system.
+
+### Used modeling tool
+[StarUML](http://staruml.io/) 
+
+## Class diagram									
+Author(s): `Teona Moga`
+
+![](imgs/classdi.jpg)
+
+The system we designed for the Fantasy Soccer project consists of 3 packages: *models*, *views* and *controllers* and one Main class. Packages *views* and *controllers* deal with the GUI part of the project and are implemented using the javafx library, such that they will be described more briefly than *models*, which is where the entire back-end logic lies.   
+### *Models* package  
+  
+This package implements 5 distinct classes: **Game**, **League**, **Team**, **Player** and **Match**, described below.  
+  
+#### **Game**  
```

Comment:
Great make sure to however keep this class focused on being an interface for the view classes. 
Example : When implementing leader boards make sure you they are generated within League and then passed on through the API, subsequently your method generateLeaderboard() is misnamed, most methods here should be mere getters and setters!